src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -178,6 +197,42 @@ private CompletableFuture<Result<LoanAndRelatedRecords>> saveCheckOutSessionReco
         succeeded(records)));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> acquireLock(LoanAndRelatedRecords records, CheckOutLockRepository checkOutLockRepository, StringBuilder checkOutLockId) {
+    log.debug("acquireLock:: Creating checkout lock {} ",records.getCheckoutLockConfiguration());
+    if(records.getCheckoutLockConfiguration() == null || !records.getCheckoutLockConfiguration().isCheckOutLockFeatureEnabled()) {
+        return completedFuture(Result.succeeded(records));
+      }
+    CompletableFuture<CheckOutLock> future = new CompletableFuture<>();
+    checkOutLockRepository.createLockWithRetry(0, future, records);
+    return future.handle((res,err) ->{
+        if(res!=null){
```

Comment:
I'm not loving checking not null here as the signal that a lock has been acquired. Instead check if `err` is not null and then handle the error. Everything you need to process the err, including the message should be in err.

Reply:
Changes made as per the suggestion. Now we are doing null check for err object.